presto-docs/src/main/sphinx/release/release-342.md

```diff
@@ -0,0 +1,77 @@
+
+## Hive Connector Changes
+
+* Add support for S3 encrypted files. ({issue}`2536`)
+* Add support for ABFS OAuth authentication. ({issue}`5052`)
+* Support reading timestamp with microsecond or nanosecond precision. This can be enabled with `hive.timestamp-precision`
+  connector configuration property. ({issue}`4953`)
+* Allow specifying what happens if data is inserted into existing Hive partition. 
+  This can be done using `hive.insert-existing-partitions-behavior` config property. ({issue}`4999`)
+* Add property (`hive.dynamic-filtering-probe-blocking-timeout`) for delaying table scans
+  until dynamic filtering can be performed more efficiently. This can also be set via the `dynamic_filtering_probe_blocking_timeout`
+  session property. ({issue}`4991`)
+* Disable matching the existing user and group of the table or partition when creating new files on HDFS.
+  The functionality was added in 341 and is now disabled by default. You can enable it with `hive.fs.new-file-inherit-ownership`
+  configuration property. ({issue}`5187`)
+* Improve performance of reading small file in RCFile format. ({issue}`2536`)
+* Improve planning time for queries with non-equality filters on
+  partition columns when using the Glue metastore. ({issue}`5060`)
+* Improve performance when reading `JSON` and `CSV` file formats. ({issue}`5142`)
+    
+# Iceberg Connector Changes
+
+* Fix partition transforms for temporal columns for dates before 1970. ({issue}`5273`)
+
+## Kafka Connector Changes
+
+* Expose message headers as a `_headers` column of `map(varchar, array(varbinary))` type. ({issue}`4462`)
+* Add write support for `TIME`, `TIME WITH TIME ZONE`, `TIMESTAMP` and `TIMESTAMP WITH TIME ZONE` 
+  for Kafka connector when using the JSON encoder. ({issue}`4743`)
+* In JSON decoder drop decoding support for temporal types for nonsensical combinations of input-format-type/data-type.
+  Following combination are no longer supported:
+  - `rfc2822`:  `DATE`, `TIME`, `TIME WITH TIME ZONE`
+  - `milliseconds-since-epoch`: `TIME WITH TIME ZONE`, `TIMESTAMP WITH TIME ZONE`
+  - `seconds-since-epoch`: `TIME WITH TIME ZONE`, `TIMESTAMP WITH TIME ZONE`
+  ({issue}`4743`)
+  
+## MySQL Connector Changes
+
+* Improve performance of `INSERT` statement when MySQL instance isn't running in GTID mode. ({issue}`4995`)
+
+## PostgreSQL Connector Changes
+
+* Extend type mapping to support variable-precision `TIMESTAMP` and `TIMESTAMP WITH TIME ZONE` types. ({issue}`5124`, {issue}`5105`)
+
+## SQL Server Connector Changes
+
+* Fix failure when inserting `NULL` to `VARBINARY` column. ({issue}`4846`)
+* Improve performance of aggregation queries by computing aggregations within SQL Server database.
+  Currently, the following aggregate functions are eligible for pushdown:
+  ``count``,  ``min``, ``max``, ``sum`` and ``avg``. ({issue}`4139`)
+
+## SPI Changes
+
+* Add `DynamicFilter#isAwaitable` method that returns whether dynamic filter is not complete and can be
```

Comment:
`DynamicFilter.isAwaitable()`

Reply:
Add `DynamicFilter.isAwaitable()` method that returns whether or not the dynamic filter is complete and can be awaited for using the `isBlocked()` method.